server/server.js

```diff
@@ -342,6 +346,151 @@ app.post('/message', jsonBodyParser, async (req, res) => {
     }
 });
 
+app.post('/heartbeat', jsonBodyParser, async (req, res) => {
+    log(`got a heartbeat from ${req.body.system_id}, flic_last_seen_secs is ${req.body.flic_last_seen_secs}, flic_last_ping_secs is ${req.body.flic_last_ping_secs}`)
+    let heartbeat = await db.getHeartbeatWithSystemId(req.body.system_id)
+    heartbeat.flicLastSeenTime = moment().subtract(req.body.flic_last_seen_secs, 'seconds').toISOString()
+    heartbeat.flicLastPingTime = moment().subtract(req.body.flic_last_ping_secs, 'seconds').toISOString()
+    heartbeat.heartbeatLastSeenTime = moment().toISOString()
+    await db.saveHeartbeat(heartbeat)
+    res.status(200).send()
+})
+
+
+app.post('/rename_system', jsonBodyParser, async (req, res) => {
+    log('got a request to rename system ' + req.body.system_id)
+    let heartbeat = await db.getHeartbeatWithSystemId(req.body.systemid)
+    heartbeat.systemName = req.body.system_name
+    await db.saveHeartbeat(heartbeat)
+    res.status(200).send()
+})
+
+app.post('/hide_system', jsonBodyParser, async (req, res) => {
+    log('got a request to hide system ' + req.body.system_id) 
+    let heartbeat = await db.getHeartbeatWithSystemId(req.body.systemid)
+    heartbeat.hidden = true
+    await db.saveHeartbeat(heartbeat)
+    res.status(200).send()
+})
+
+app.post('/unhide_system', jsonBodyParser, async (req, res) => {
+    log('got a request to hide system ' + req.body.system_id) 
+    let heartbeat = await db.getHeartbeatWithSystemId(req.body.systemid)
+    heartbeat.hidden = false
+    await db.saveHeartbeat(heartbeat)
+    res.status(200).send()
+})
+
+
+app.post('/mute_system', jsonBodyParser, async (req, res) => {
+    log('got a request to mute system ' + req.body.system_id) 
+    let heartbeat = await db.getHeartbeatWithSystemId(req.body.systemid)
+    heartbeat.muted = true
+    await db.saveHeartbeat(heartbeat)
+    res.status(200).send()
+})
+
+app.post('/unmute_system', jsonBodyParser, async (req, res) => {
+    log('got a request to unmute system ' + req.body.system_id) 
+    let heartbeat = await db.getHeartbeatWithSystemId(req.body.systemid)
+    heartbeat.muted = false
+    await db.saveHeartbeat(heartbeat)
+    res.status(200).send()
+})
+
+app.get('/heartbeatDashboard', async (req, res) => {
+    let heartbeats = await db.getHeartbeats()
+    let viewParams = {
+        domain: helpers.getEnvVar('DOMAIN'),
+        dashboard_render_time: moment().toString(),
+        systems: []   
+    }
+
+    for(const heartbeat of heartbeats){
+
+        if(heartbeat.hidden) {
+            return
```

Comment:
This should be `break` so that we will still see the other non-hidden heartbeats that come after this one in the list.

Reply:
this should also be the case for the function checkHeartbeat() right?

Reply:
Yes! Good catch.